plugin/depsusage/depsusage.go

```diff
@@ -0,0 +1,127 @@
+package depsusage
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/safedep/code/core"
+	"github.com/safedep/code/pkg"
+	"github.com/safedep/dry/log"
+	sitter "github.com/smacker/go-tree-sitter"
+)
+
+type dependencyUsagePlugin struct {
+	// Callback function which is called with the usage evidence
+	usageCallback func(evidence *UsageEvidence)
+}
+
+// Verify contract
+var _ core.TreePlugin = (*dependencyUsagePlugin)(nil)
+
+func NewDependencyUsagePlugin(usageCallback func(evidence *UsageEvidence)) *dependencyUsagePlugin {
+	return &dependencyUsagePlugin{
+		usageCallback: usageCallback,
+	}
+}
+
+func (p *dependencyUsagePlugin) Name() string {
+	return "DependencyUsagePlugin"
+}
+
+var supportedLanguages = []core.LanguageCode{core.LanguageCodePython}
+
+func (p *dependencyUsagePlugin) SupportedLanguages() []core.LanguageCode {
+	return supportedLanguages
+}
+
+func (p *dependencyUsagePlugin) AnalyzeTree(ctx context.Context, tree core.ParseTree) error {
+	lang, err := tree.Language()
+	if err != nil {
+		return fmt.Errorf("failed to get language: %w", err)
+	}
+
+	file, err := tree.File()
+	if err != nil {
+		return fmt.Errorf("failed to get file: %w", err)
+	}
+
+	log.Debugf("depsusage - Analyzing tree for language: %s, file: %s\n",
+		lang.Meta().Code, file.Name())
+
+	imports, err := lang.Resolvers().ResolveImports(tree)
+	if err != nil {
+		return fmt.Errorf("failed to resolve imports: %w", err)
+	}
+
+	moduleIdentifiers := make(map[string]*identifierItem)
+	for _, imp := range imports {
+		baseModuleName := pkg.GetBaseModuleName(imp.ModuleName())
```

Comment:
This is pretty much language specific i.e. `GetBaseModuleName` will depend on language. While for Python `x.y` will return `x`, for a language like Go where import will be something like `github.com/x/y/z`, different rules are required to resolve base name of an import. 

This need some thinking to model it properly. For now, I suggest use a local function and do not introduce a re-usable concept or function like `pkg.GetBaseModuleName`

I feel we need to define a contract for languages to implement module name resolvers. But lets look at it as a future task.

Reply:
makes sense, moved inside plugin for now